python_modules/dagster-graphql/dagster_graphql/schema/instigation.py

```diff
@@ -169,87 +171,110 @@ def resolve_logEvents(self, graphene_info: ResolveInfo):
+                )
+            except Exception:
+                sensor_data = serializable_error_info_from_exc_info(sys.exc_info())
+            return GrapheneTickEvaluation(sensor_data)
+        else:
+            if not repository.has_external_schedule(self._selector.schedule_name):
+                return None
+            if not self.timestamp:
+                return None
+            external_schedule = repository.get_external_schedule(self._selector.schedule_name)
+            timezone_str = external_schedule.execution_timezone
+            if not timezone_str:
+                timezone_str = "UTC"
+
+            next_tick_datetime = next(external_schedule.execution_time_iterator(self.timestamp))
+            schedule_time = to_timezone(pendulum.instance(next_tick_datetime), timezone_str)
+            schedule_data: Union[ScheduleExecutionData, SerializableErrorInfo]
+            try:
+                schedule_data = repository_location.get_external_schedule_execution_data(
+                    instance=graphene_info.context.instance,
+                    repository_handle=repository.handle,
+                    schedule_name=external_schedule.name,
+                    scheduled_execution_time=schedule_time,
+                )
+            except Exception:
+                schedule_data = serializable_error_info_from_exc_info(sys.exc_info())
 
-        return GrapheneTickEvaluation(schedule_data)
+            return GrapheneTickEvaluation(schedule_data)
 
 
 class GrapheneTickEvaluation(graphene.ObjectType):
     runRequests = graphene.List(lambda: GrapheneRunRequest)
     skipReason = graphene.String()
     error = graphene.Field(GraphenePythonError)
+    cursor = graphene.String()
 
     class Meta:
         name = "TickEvaluation"
 
-    def __init__(self, schedule_data):
+    def __init__(self, execution_data):
         check.inst_param(
-            schedule_data,
-            "schedule_data",
-            (ScheduleExecutionData, SerializableErrorInfo),
+            execution_data,
+            "execution_data",
+            (ScheduleExecutionData, SensorExecutionData, SerializableErrorInfo),
         )
         error = (
-            GraphenePythonError(schedule_data)
-            if isinstance(schedule_data, SerializableErrorInfo)
+            GraphenePythonError(execution_data)
+            if isinstance(execution_data, SerializableErrorInfo)
             else None
         )
         skip_reason = (
-            schedule_data.skip_message if isinstance(schedule_data, ScheduleExecutionData) else None
+            execution_data.skip_message
+            if isinstance(execution_data, ScheduleExecutionData)
+            else None
         )
         self._run_requests = (
-            schedule_data.run_requests if isinstance(schedule_data, ScheduleExecutionData) else None
+            execution_data.run_requests
+            if isinstance(execution_data, ScheduleExecutionData)
+            else None
         )
```

Comment:
why are skip reason and run requests limited to schedules? can't you get these on sensors too?

Reply:
whoops good catch - will add this to test cases for sensors too